ddl/index_cop.go

```diff
@@ -401,7 +413,7 @@ func (c *copContext) fetchTableScanResult(ctx context.Context, result distsql.Se
 		return nil, false, errors.Trace(err)
 	}
 	if chk.NumRows() == 0 {
-		return buf, true, nil
```

Comment:
Here means finished scan, we closed result to release mem usage? how about   before err return should we also have safeguard to release mem?